stake-pool/program/src/processor.rs

```diff
@@ -291,32 +313,99 @@ impl Processor {
             .ok_or(Error::CalculationFailure)?;
         let pool_amount = <u64>::try_from(pool_amount).or(Err(Error::CalculationFailure))?;
 
+        let withdraw_bump_seed = [stake_pool.withdraw_bump_seed];
+        let stake_pool_key_bytes = stake_pool_info.key.to_bytes();
+        let withdraw_signers = prepare_signers!(
+            stake_pool_key_bytes,
+            Self::AUTHORITY_WITHDRAW,
+            withdraw_bump_seed
+        );
+
         Self::stake_split(
-            stake_pool_info.key,
             source_info.clone(),
             withdraw_info.clone(),
-            stake_pool.withdraw_bump_seed,
             stake_amount,
             stake_info.clone(),
+            withdraw_signers,
         )?;
 
         Self::stake_authorize(
-            stake_pool_info.key,
             stake_info.clone(),
             withdraw_info.clone(),
-            stake_pool.withdraw_bump_seed,
             dest_user_info.key,
             stake::StakeAuthorize::Withdrawer,
+            withdraw_signers,
         )?;
 
         Self::token_burn(
-            stake_pool_info.key,
             token_program_info.clone(),
             source_info.clone(),
             pool_mint_info.clone(),
             withdraw_info.clone(),
-            stake_pool.withdraw_bump_seed,
             pool_amount,
+            withdraw_signers,
+        )?;
+
+        stake_pool.pool_total -= pool_amount;
+        stake_pool.stake_total -= stake_amount;
+        State::Init(stake_pool).serialize(&mut stake_pool_info.data.borrow_mut())?;
+        Ok(())
+    }
+    /// Processes an [Claim](enum.Instruction.html).
+    pub fn process_claim(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let stake_pool_info = next_account_info(account_info_iter)?;
+        let withdraw_info = next_account_info(account_info_iter)?;
+        let source_info = next_account_info(account_info_iter)?;
```

Comment:
You'll need an additional parameter of the user's pool token account to burn from.  `source_info` is currently referenced as both the stake account to reassign *and also* the pool token account to burn from

Reply:
Yes, this is a bug, I believe there was a similar bug in the original split method, will fix both